cmd/minikube/cmd/start_flags.go

```diff
@@ -560,6 +560,13 @@ func generateNewConfigFromFlags(cmd *cobra.Command, k8sVersion string, rtime str
 		out.WarningT("--network flag is only valid with the docker/podman, qemu, kvm, and vfkit drivers, it will be ignored")
 	}
 
+	// Set the cluster dns name to the dnsDomain flag. If the flag is not set, set it to
+	// the configured default.
+	defaultClusterDNSDomain := viper.GetString(dnsDomain)
+	if defaultClusterDNSDomain == constants.DefaultClusterDNSDomain {
+		defaultClusterDNSDomain = viper.GetString(config.DefaultClusterDNSDomain)
+	}
```

Comment:
The rename will make this much more clear:

```go
if dnsDomain == constants.DefaultClusterDNSDomain {
    dnsDomain = viper.GetString(config.DefaultClusterDNSDomain)
}
```

Using 2 variables called defaultXXX made this code so confusing that we added a comment.